worlds/witness/items.py

```diff
@@ -33,214 +39,221 @@ class StaticWitnessItems:
+                or is_option_enabled(self._world, self._player_id, "shuffle_symbols"):
+            if get_option_value(self._world, self._player_id, "shuffle_doors") > 0:
+                output = ["Dots", "Black/White Squares", "Symmetry"]
+            else:
+                output = ["Dots", "Black/White Squares", "Symmetry", "Shapers", "Stars"]
 
-            if is_option_enabled(multiworld, player, "shuffle_discarded_panels"):
-                if get_option_value(multiworld, player, "puzzle_randomization") == 1:
-                    self.GOOD_ITEMS.append("Arrows")
+            if is_option_enabled(self._world, self._player_id, "shuffle_discarded_panels"):
+                if get_option_value(self._world, self._player_id, "puzzle_randomization") == 1:
+                    output.append("Arrows")
                 else:
-                    self.GOOD_ITEMS.append("Triangles")
-
-            self.GOOD_ITEMS = [
-                StaticWitnessLogic.ITEMS_TO_PROGRESSIVE.get(item, item) for item in self.GOOD_ITEMS
-            ]
-
-        for event_location in locat.EVENT_LOCATION_TABLE:
-            location = logic.EVENT_ITEM_PAIRS[event_location]
-            self.EVENT_ITEM_TABLE[location] = ItemData(None, True, True)
-            self.ITEM_TABLE[location] = ItemData(None, True, True)
-
-        trap_percentage = get_option_value(multiworld, player, "trap_percentage")
-
-        self.JUNK_WEIGHTS = dict()
-
-        if trap_percentage != 0:
-            # I'm sure there must be some super "pythonic" way of doing this :D
-
-            for trap_name, trap_weight in StaticWitnessItems.TRAP_WEIGHTS.items():
-                self.JUNK_WEIGHTS[trap_name] = (trap_weight * trap_percentage) / 100
-
-        if trap_percentage != 100:
-            for bonus_name, bonus_weight in StaticWitnessItems.BONUS_WEIGHTS.items():
-                self.JUNK_WEIGHTS[bonus_name] = (bonus_weight * (100 - trap_percentage)) / 100
-
-        self.JUNK_WEIGHTS = {
-            key: value for (key, value)
-            in self.JUNK_WEIGHTS.items()
-            if key in self.ITEM_TABLE.keys()
-        }
-
-        # JUNK_WEIGHTS will add up to 1 if the boosts weights and the trap weights each add up to 1 respectively.
-
-        for junk_item in StaticWitnessItems.ALL_JUNK_ITEMS:
-            if junk_item not in self.JUNK_WEIGHTS.keys():
-                del self.ITEM_TABLE[junk_item]
+                    output.append("Triangles")
+
+            # Replace progressive items with their parents.
+            output = [StaticWitnessLogic.get_parent_progressive_item(item) for item in output]
+
+        # Remove items that are mentioned in any plando options. (Hopefully, in the future, plando will get resolved
+        #   before create_items so that we'll be able to check placed items instead of just removing all items mentioned
+        #   regardless of whether or not they actually wind up being manually placed.
```

Comment:
I tried moving this code to pre_fill which is after plando gets resolved but an AP core dev wasn't happy with my implementation because it performed an "in" operation on the whole multiworld's itempool.

There may be a better way that I realized recently, by creating an actual reference of all the "good items" in create_items, then checking if they have a "parent location" by pre_fill, which would suggest they are no longer available for placement. That way, you won't have to check whether they're still in the itempool. Not sure, I might come back to this at some point

Reply:
I'll note here that this comment's been basically moved verbatim from where this whole exclude block originally was in create_items, so this is definitely not something I understand enough to make changes to.